coderd/workspaceagents.go

```diff
@@ -1603,30 +1625,25 @@ func (api *API) watchWorkspaceAgentMetadata(rw http.ResponseWriter, r *http.Requ
 	}
 	// Prevent handler from returning until the sender is closed.
 	defer func() {
-		<-senderClosed
+		<-sseSenderClosed
 	}()
 
 	const refreshInterval = time.Second * 5
 	refreshTicker := time.NewTicker(refreshInterval)
 	defer refreshTicker.Stop()
 
-	var (
-		lastDBMetaMu sync.Mutex
-		lastDBMeta   []database.WorkspaceAgentMetadatum
-	)
+	var lastMetadata []database.WorkspaceAgentMetadatum
 
-	sendMetadata := func(pull bool) {
-		log.Debug(ctx, "sending metadata update", "pull", pull)
-		lastDBMetaMu.Lock()
-		defer lastDBMetaMu.Unlock()
+	sendMetadata := func(pullDB bool) {
+		log.Debug(ctx, "sending metadata update", "pull_db", pullDB)
 
-		var err error
-		if pull {
+		if pullDB {
+			var err error
 			// We always use the original Request context because it contains
 			// the RBAC actor.
-			lastDBMeta, err = api.Database.GetWorkspaceAgentMetadata(ctx, workspaceAgent.ID)
+			lastMetadata, err = api.Database.GetWorkspaceAgentMetadata(ctx, workspaceAgent.ID)
```

Comment:
If this fails we should probably disconnect the client. Pull only happens once and without `lastMetadata` populated to all metadata entries, we will never return any info.